### ISSUE TITLE
Paired devices are always returned by bluetooth.discover_devices() even if not in range. (msbt stack only)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [pybluez/pybluez#24]
```
What steps will reproduce the problem?
1. Use any bluetooth device that works with msbt stack
2. Pair it with another device (phone, etc) with the OS wizards. Once
paired it will appear in the Devices tab in the bluetooth settings window.
3. Turn off bluetooth on the other device (phone, etc)
4. Execute bluetooth.discover_devices()

What is the expected output? What do you see instead?
Expecting to see *only* the list of devices in range. Instead, the list
also contains all paired devices no matter if they are in range or not.

What version of the product are you using? On what operating system?
PyBluez-0.18.win32-py2.6 with python 2.6.4 tested both on Windows XP/SP3
and Windows 7

Please provide any additional information below.
Also tried flush_cache=True/False - makes no difference. If I use the
widcomm stack then only devices in range are returned (as epxected!)
```

Original issue reported on code.google.com by `htren...@gmail.com` on 3 Dec 2009 at 12:37

**Comment by karulis** [comment]
For reference:
http://stackoverflow.com/questions/18826551/get-bluetooth-devices-in-range

**Comment by karulis** [comment]
I have merged 6961b70bc1a1e9af97ab982ccd8826f560d99f7e, but this commit does not fix this issue.
It seems that there is no simple way to fix it.

For device discovery in ms stack we have following parameters:

``` c++
search_criteria.fReturnAuthenticated
search_criteria.fReturnRemembered
search_criteria.fReturnConnected
search_criteria.fReturnUnknown
search_criteria.fIssueInquiry
```

setting fIssueInquiry to True and fReturnAuthenticated(so paired) to False
will not return paired devices that are in range.

Commit 6961b70bc1a1e9af97ab982ccd8826f560d99f7e enables two parameters for windows:
- duration of inquiry
- flush_cache - False=return cached devices

**Comment by karulis** [comment]
As a workaround - You need to connect to device to check if it is in range(or do https://github.com/karulis/pybluez/blob/master/examples/simple/sdp-browse.py)
